addons/controllers/cpi/cpiconfig_controller.go

```diff
@@ -61,11 +56,11 @@ func (r *CPIConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 
 	cpiConfig = cpiConfig.DeepCopy()
 	cluster, err := r.getOwnerCluster(ctx, cpiConfig)
-	if err != nil {
-		return ctrl.Result{}, err
+	if cluster == nil {
+		return ctrl.Result{}, err // no need to requeue if cluster is not found
```

Comment:
Oops, error is left unchecked here

Reply:
it is checked. If err is not nil, then the returned cluster  will be nil.

Reply:
hmm, you seem to just check the cluster for nilness. Am I missing something?
```
	cluster, err := r.getOwnerCluster(ctx, cpiConfig)
	if cluster == nil {
		return ctrl.Result{}, err // no need to requeue if cluster is not found
	}
	if res, err := r.ReconcileCPIConfig(ctx, cpiConfig, cluster); err != nil {
		r.Log.Error(err, "Failed to reconcile CPIConfig")
```

Reply:
`getOwnerCluster` ensures that it returns cluster as nil whenever

* the cluster is not found
* other error occurs

So by checking the cluster's nilness, we are handling both scenarios above.